### DISCUSSION TITLE
Debug invalid connections

### DISCUSSION THREAD

**Discussion by impredicative** [evilsocket/opensnitch#570] [Q&A] [answered]
What is the meaning of "Debug invalid connections"? What differences and consequences can I expect with it unchecked versus checked?

**Comment by gustavo-iniguez-goya** [comment, marked as answer]
If you mark that option, and there's an outgoing connection that we cannot associate to an application, opensnitch will display a pop-up without the application information (path, cwd, binary name, etc). If you leave it unchecked, opensnitch will apply the default action configured (deny/allow).

![image](https://user-images.githubusercontent.com/2742953/146979459-4c88ae7c-37ed-4995-a761-f06de8fd6901.png)

In previous versions, sometimes we weren't able to obtain what application was trying to establish a new outgoing connection. In theses cases we displayed a pop-up with the text "Outgoing connection", showing only what IP and port an application was trying to reach.

If we ignored completely these connections, allowing or denying them, users would miss what's going on their systems. So we added an option to let the users choose what they prefer.

Nowadays, using eBPF it's rare if we miss what application is trying to establish a connection. If you experience problems with a particular application (like VPNs, NFS, samba), you can mark it and see if opensnitch prompts you to allow/deny an "Outgoing connection".

Starting from [v1.5.0rc1](https://github.com/evilsocket/opensnitch/releases/tag/v1.5.0-rc.1) we intercept more connections, so the problems we had with VPNs and other applications are fixed by now (:crossed_fingers: )
